web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -209,455 +235,459 @@ public TreeRow(int offsetInSnapshot, Object[] dataColumns, Object rowStyleColumn
+                            LazyString.of(() -> widget.getTicket().getTicket_asB64()),
+                            columnsBitset,
+                            range,
+                            alwaysFireEvent);
+                    BiDiStream<FlightData, FlightData> doExchange =
+                            connection.<FlightData, FlightData>streamFactory().create(
+                                    headers -> connection.flightServiceClient().doExchange(headers),
+                                    (first, headers) -> connection.browserFlightServiceClient().openDoExchange(first,
+                                            headers),
+                                    (next, headers, c) -> connection.browserFlightServiceClient().nextDoExchange(next,
+                                            headers,
+                                            c::apply),
+                                    new FlightData());
+
+                    FlightData subscriptionRequestWrapper = new FlightData();
+                    Builder doGetRequest = new Builder(1024);
+                    double columnsOffset = BarrageSubscriptionRequest.createColumnsVector(doGetRequest,
+                            makeUint8ArrayFromBitset(columnsBitset));
+                    double viewportOffset = BarrageSubscriptionRequest.createViewportVector(doGetRequest,
+                            serializeRanges(
+                                    Collections.singleton(
+                                            range)));
+                    double serializationOptionsOffset = BarrageSubscriptionOptions
+                            .createBarrageSubscriptionOptions(doGetRequest, ColumnConversionMode.Stringify, true,
+                                    updateInterval, 0, 0);
+                    double tableTicketOffset =
+                            BarrageSubscriptionRequest.createTicketVector(doGetRequest,
+                                    viewTicket.ticket.getTicket_asU8());
+                    BarrageSubscriptionRequest.startBarrageSubscriptionRequest(doGetRequest);
+                    BarrageSubscriptionRequest.addTicket(doGetRequest, tableTicketOffset);
+                    BarrageSubscriptionRequest.addColumns(doGetRequest, columnsOffset);
+                    BarrageSubscriptionRequest.addSubscriptionOptions(doGetRequest, serializationOptionsOffset);
+                    BarrageSubscriptionRequest.addViewport(doGetRequest, viewportOffset);
+                    doGetRequest.finish(BarrageSubscriptionRequest.endBarrageSubscriptionRequest(doGetRequest));
+
+                    subscriptionRequestWrapper.setAppMetadata(
+                            WebBarrageUtils.wrapMessage(doGetRequest, BarrageMessageType.BarrageSubscriptionRequest));
+                    doExchange.send(subscriptionRequestWrapper);
+
+                    // hang on to this to get server-sent snapshots, cancel when needed
+                    // doExchange.end();
```

Comment:
Seems you don't need the `// doExchagne.end();` line here
```suggestion
                    // hang on to the doExchange stream to get server-sent snapshots, cancel when needed
                    // doExchange.end();
```